### PR TITLE
🎉 Source Github: Add MultipleTokenAuthenticator

### LINKED ISSUE #3250: Source Github: Allow rotating between multiple API keys
## Tell us about the problem you're trying to solve
Github has stringent API key limits. Users have asked to have the connector supply providing multiple API keys and then have the connector rotate between them. 

The fastest way to do this would be to implement it directly in the connector. But we can also make this available to any connector by adding this as an authenticator in the CDK.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368165169868) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by gaart**
## What
Multiple token support added to the CDK.
Solving airbytehq/airbyte#3250

## How
Now we have MultipleTokenAuthenticator which receives a comma-separated list of tokens and rotates them during API calls.

### CODE REVIEW

**Review by vitaliizazmic — COMMENTED** [review]
Could you please explain how token rotating is implemented while exceeded key limits?

**Comment by gaart** [comment]
@vitaliizazmic 
> Could you please explain how token rotating is implemented while exceeded key limits?

There's a `backoff_time` mechanism which adds timeouts when the limits exceeded, this PR doesn't change that behaviour but allows to use multiple tokens rotating them after each API call

**Comment by vitaliizazmic** [comment]
@gaart sounds good. Don't forget to run tests and format.

**Comment by lukebussey** [comment]
Why pass a comma separated list of tokens when you can just pass over a List?

File: airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/token.py [hunk 1]
```diff
@@ -36,3 +42,14 @@ def __init__(self, token: str, auth_method: str = "Bearer", auth_header: str = "
 
     def get_auth_header(self) -> Mapping[str, Any]:
         return {self.auth_header: f"{self.auth_method} {self._token}"}
+
+
+class MultipleTokenAuthenticator(HttpAuthenticator):
+    def __init__(self, tokens: str, auth_method: str = "Bearer", auth_header: str = "Authorization"):
```
Comment (sherifnada) [inline airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/token.py:48]: this should take a list of strings rather than a comma separated string

**Review by sherifnada — CHANGES_REQUESTED** [review]
one requested change but otherwise looks great

File: airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/token.py (hunk 1 above)
Comment (gaart) [reply airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/token.py]: good point, updated

File: airbyte-cdk/python/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## 0.1.10
+Upd multiple token support: switch to list of tokens
+
+## 0.1.9
```
Comment (sherifnada) [inline airbyte-cdk/python/CHANGELOG.md:6]: do we need two versions for the same release? can we just make it a single version?

File: airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/token.py [hunk 3]
```diff
@@ -36,3 +37,14 @@ def __init__(self, token: str, auth_method: str = "Bearer", auth_header: str = "
 
     def get_auth_header(self) -> Mapping[str, Any]:
         return {self.auth_header: f"{self.auth_method} {self._token}"}
+
+
+class MultipleTokenAuthenticator(HttpAuthenticator):
```
Comment (sherifnada) [inline airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/token.py:42]: could you add a doc string explaining what this does e.g: 

```
Uses the input list of tokens for authentication in a round-robin fashion. This allows load balancing quota consumption across multiple tokens. 
```

File: airbyte-integrations/connectors/source-github/source_github/spec.json [hunk 4]
```diff
@@ -9,7 +9,7 @@
     "properties": {
       "access_token": {
         "type": "string",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-github/source_github/spec.json:11]:
```suggestion
        "type": "string",
        "title": "Access Tokens",
```

File: airbyte-integrations/connectors/source-github/source_github/spec.json [hunk 5]
```diff
@@ -9,7 +9,7 @@
...
-        "description": "Log into Github and then generate a <a href=\"https://github.com/settings/tokens\"> personal access token</a>.",
+        "description": "Log into Github and then generate a <a href=\"https://github.com/settings/tokens\"> personal access token</a>. Separate multiple tokens with \",\"",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-github/source_github/spec.json:12]: Adding some more context to why someone might want to use multiple tokens. 
```suggestion
        "description": "Log into Github and then generate a <a href=\"https://github.com/settings/tokens\"> personal access token</a>. To load balance your API quota consumption across multiple API tokens, input multiple tokens separated with \",\"",
```

Comment (sherifnada) [inline airbyte-integrations/connectors/source-github/source_github/spec.json:12]: Ideally, `type: string` would be an array of strings too. However for backwards compatibility this is not great. 

Could you create an issue to update this to a list of strings once #5396 is resolved?

File: airbyte-integrations/connectors/source-github/source_github/source.py [hunk 6]
```diff
@@ -65,28 +70,41 @@ def _generate_repositories(self, config: Mapping[str, Any], authenticator: Token
...
+                    r["full_name"] for r in repos.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream)
                 ]
         if repositories:
             repositories_list += repositories
 
         return list(set(repositories_list))
 
+    @staticmethod
+    def _get_authenticator(token: str):
+        if TOKEN_SEPARATOR not in token:
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-github/source_github/source.py:82]: do we need a special edge case for this? multiple auth should still work with one token right?

**Review by sherifnada — APPROVED** [review]
Added some requested improvements, feel free to merge once they are addressed!

File: airbyte-cdk/python/CHANGELOG.md (hunk 2 above)
Comment (gaart) [reply airbyte-cdk/python/CHANGELOG.md:6]: Probably not. But both are released already

**Pull request merged**